framework/include/media/HttpInputDataSource.h

```diff
@@ -0,0 +1,147 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+/**
+ * @ingroup MEDIA
+ * @{
+ */
+
+/**
+ * @file media/HttpInputDataSource.h
+ * @brief Media HttpInputDataSource APIs
+ */
+
+#ifndef __MEDIA_HTTPINPUTDATASOURCE_H
+#define __MEDIA_HTTPINPUTDATASOURCE_H
+
+#include <media/InputDataSource.h>
+#include <curl/curl.h>
+#include <curl/easy.h>
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <string>
+
+namespace media {
+namespace stream {
+
+#ifndef CONFIG_NET
```

Comment:
If we are using only features of libCurl, I think we can check only CONFIG_ENABLE_CURL in httpStream class.

Reply:
Fixed